src/main/java/org/testng/SuiteRunner.java

```diff
@@ -792,7 +793,7 @@ public void beforeInvocation(IInvokedMethod method, ITestResult testResult) {
 
   @Override
   public List<IInvokedMethod> getAllInvokedMethods() {
-    return invokedMethods;
+    return new ArrayList<>(invokedMethods);
```

Comment:
Why not `Collections.unmodifiableList()`?

Reply:
Same for the other locations.

Reply:
For me to do that, I would have to do something like below

`return Collections.unmodifiableList(new ArrayList<>(invokedMethods));`

because invokedMethods is now a `Collection` and not a `List`. Do you think we need to be doing all that ?

Reply:
Ok, it is out of the scope of the current PR. But unmodifiable collections coud help to detect weird behavior.